prov/efa/test/efa_unit_test_cq.c

```diff
@@ -74,111 +74,88 @@ void test_ibv_cq_ex_read_empty_cq(struct efa_resource **state)
  * then user should call fi_cq_readerr() to get an error CQ entry that contain error code.
  *
  * @param[in]  state            struct efa_resource that is managed by the framework
- * @param[in]  ep_type          Endpoint type, can be FI_EP_DGRAM or FI_EP_RDM
  * @param[in]  local_host_id    Local(sender) host id
  * @param[in]  peer_host_id     Peer(receiver) host id
  * @param[in]  vendor_error     Vendor error returned by ibv_read_vendor_err
  * @param[in]  err_prefix       Expected error message prefix from fi_cq_strerror. For RDM endpoint the error
  *                              might contain conditional information after the prefix.
  */
-static void test_cq_read_bad_send_status(struct efa_resource *resource, enum fi_ep_type ep_type,
-                                         uint64_t local_host_id, uint64_t peer_host_id,
-                                         int vendor_error, const char *err_prefix)
+static void test_rdm_cq_read_bad_send_status(struct efa_resource *resource,
+                                             uint64_t local_host_id, uint64_t peer_host_id,
+                                             int vendor_error, const char *err_prefix)
 {
 	const char *strerror;
 	fi_addr_t addr;
 	int ret, err;
+	char host_id_str[] = "xxxxx host id: i-01234567812345678";
 	size_t raw_addr_len = sizeof(struct efa_ep_addr);
 	struct efa_ep_addr raw_addr;
 	struct efa_unit_test_buff send_buff;
 	struct fi_cq_data_entry cq_entry;
 	struct fi_cq_err_entry cq_err_entry = {0};
 	struct ibv_cq_ex *ibv_cqx;
-	struct ibv_qp *ibv_qp;
+	struct ibv_qp_ex *ibv_qpx;
 	struct efa_rdm_ep *efa_rdm_ep;
+	struct efa_rdm_peer *peer;
 
-	efa_unit_test_resource_construct(resource, ep_type);
+	efa_unit_test_resource_construct(resource, FI_EP_RDM);
 	efa_unit_test_buff_construct(&send_buff, resource, 4096 /* buff_size */);
 
-	if (ep_type == FI_EP_RDM) {
-		efa_rdm_ep = container_of(resource->ep, struct efa_rdm_ep, base_ep.util_ep.ep_fid);
-
-		efa_rdm_ep->host_id = local_host_id;
-
-		ibv_qp =  efa_rdm_ep->base_ep.qp->ibv_qp;
+	efa_rdm_ep = container_of(resource->ep, struct efa_rdm_ep, base_ep.util_ep.ep_fid);
+	efa_rdm_ep->host_id = local_host_id;
+	ibv_qpx = efa_rdm_ep->base_ep.qp->ibv_qp_ex;
+	ibv_cqx = efa_rdm_ep->ibv_cq_ex;
+	/* set use_shm_for_tx to false to force efa_rdm_ep to use efa device to send,
+	 * which means call ibv_post_send
```

Comment:
I searched `ibv_post_send` on this page there are a few places. I think you intended to remove them.

Reply:
I removed the reference to `ibv_post_send` in the other PR https://github.com/ofiwg/libfabric/pull/9058. Will rebase this PR once that one is merged